packages/@react-spectrum/list/test/ListView.test.js

```diff
@@ -598,4 +595,408 @@ describe('ListView', function () {
+        expect(await onDrop.mock.calls[0][0].items.length).toBe(3);
+        expect(await onDrop.mock.calls[0][0].items[0].getText('text/plain')).toBe('Item a');
+        expect(await onDrop.mock.calls[0][0].items[1].getText('text/plain')).toBe('Item b');
+        expect(await onDrop.mock.calls[0][0].items[2].getText('text/plain')).toBe('Item d');
+
+        expect(onDragEnd).toHaveBeenCalledTimes(1);
+        expect(onDragEnd).toHaveBeenCalledWith({
+          type: 'dragend',
+          keys: new Set(['a', 'b', 'd']),
+          x: 50,
+          y: 25,
+          dropOperation: 'move'
+        });
+      });
+    });
+
+    it.skip('should toggle selection upon clicking the row checkbox', function () {
+      // TODO: this is currently broken in draggable listviews
+    });
+
+    it.skip('should open a menu upon click', function () {
+      // TODO: this is currently broken in draggable listviews
+    });
+
+    it('should not display the drag handle on hover, press, or keyboard focus for disabled/non dragggable items', function () {
+      let itemAllowsDragging = (key) => {
+        if (key === 'b') {
+          return false;
+        }
+        return true;
+      };
+
+      let {getAllByRole} = render(
+        <DraggableListView itemAllowsDragging={itemAllowsDragging} disabledKeys={['a']} />
+      );
+
+      let rows = getAllByRole('row');
+      let cellA = within(rows[0]).getByRole('gridcell');
+      let cellB = within(rows[1]).getByRole('gridcell');
+      let cellC = within(rows[2]).getByRole('gridcell');
+
+      act(() => cellA.focus());
+      expect(within(cellA).queryByTestId('draghandle')).toBeFalsy();
+      moveFocus('ArrowDown');
+      expect(within(cellB).queryByTestId('draghandle')).toBeFalsy();
+      moveFocus('ArrowDown');
+      expect(within(cellC).queryByTestId('draghandle')).toBeTruthy();
+      moveFocus('ArrowDown');
+      expect(within(cellC).queryByTestId('draghandle')).toBeFalsy();
+
+      fireEvent.mouseDown(cellA, {detail: 1});
+      expect(within(cellA).queryByTestId('draghandle')).toBeFalsy();
+      fireEvent.mouseDown(cellB, {detail: 1});
+      expect(within(cellB).queryByTestId('draghandle')).toBeFalsy();
+      fireEvent.mouseDown(cellC, {detail: 1});
+      expect(within(cellC).queryByTestId('draghandle')).toBeTruthy();
+      fireEvent.mouseUp(cellC, {detail: 1});
+      expect(within(cellC).queryByTestId('draghandle')).toBeFalsy();
+
+      // TODO: for some reason this doesn't seem to be triggering the hover listeners properly...
+      // fireEvent.mouseEnter(cellA);
+      // expect(within(cellA).queryByTestId('draghandle')).toBeFalsy();
+      // fireEvent.mouseEnter(cellB);
+      // expect(within(cellB).queryByTestId('draghandle')).toBeFalsy();
+      // fireEvent.mouseEnter(cellC);
+      // expect(within(cellC).queryByTestId('draghandle')).toBeTruthy();
+      // fireEvent.mouseLeave(cellC);
+      // expect(within(cellC).queryByTestId('draghandle')).toBeFalsy();
```

Comment:
This test passes if I add any arbitrary onMouseEnter listener to the row element, not entirely sure why...

Reply:
thanks for catching this comment, I forgot all about it haha. Funnily enough it works now ¯\_(ツ)_/¯, albeit with a modified cell C check since that should be a non-draggable cell. Dunno what past me got stuck with...